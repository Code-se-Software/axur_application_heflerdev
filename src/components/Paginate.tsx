import React, {useEffect, useState} from 'react';
import ReactPaginate from 'react-paginate';

interface Paginated {
    itemsPerPage: number
    items: Array<string | number>
}

const Items = ({currentItems}: { currentItems: Array<string> }) => {
    return (
        <>
            {currentItems &&
                currentItems.map((item) => (
                    <div>
                        <h3>Item #{item}</h3>
                    </div>
                ))}
        </>
    );
}

const PaginatedItems = ({itemsPerPage, items}: Paginated) => {
    const [currentItems, setCurrentItems] = useState(null);
    const [pageCount, setPageCount] = useState(0);
    const [itemOffset, setItemOffset] = useState(0);

    useEffect(() => {
        const endOffset = itemOffset + itemsPerPage;
        setCurrentItems(items.slice(itemOffset, endOffset));
        setPageCount(Math.ceil(items.length / itemsPerPage));
    }, [itemOffset, itemsPerPage]);

    const handlePageClick = (event: { selected: number; }) => {
        const newOffset = (event.selected * itemsPerPage) % items.length;
        console.log(
            `User requested page number ${event.selected}, which is offset ${newOffset}`
        );
        setItemOffset(newOffset);
    };

    return (
        <>
            <Items currentItems={currentItems}/>
            <ReactPaginate
                breakLabel="..."
                nextLabel="next >"
                onPageChange={handlePageClick}
                pageRangeDisplayed={5}
                pageCount={pageCount}
                previousLabel="< previous"
                renderOnZeroPageCount={null}
            />
        </>
    );
}

export default PaginatedItems;