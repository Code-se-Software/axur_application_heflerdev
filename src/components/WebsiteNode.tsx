import React from "react";
import {Row} from "react-bootstrap";

const WebsiteNode = () => {
    return (
        <Row>

        </Row>
    )
}

export default WebsiteNode;
