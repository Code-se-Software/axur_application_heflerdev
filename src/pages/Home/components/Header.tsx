import React from "react";
import {Container, Stack} from "react-bootstrap";
import {UnicornIcon} from "../../../assets/icons/Icons";

export const Header = () => (
    <Container>
        <Stack className={"home-header"}>
            <div className={"icon"}><UnicornIcon size={78}/></div>
            <div className={"text"}>UniSearcher</div>
        </Stack>
    </Container>
)