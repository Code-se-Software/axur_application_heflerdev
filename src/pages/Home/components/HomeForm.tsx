import React, {Dispatch, SetStateAction, useEffect, useState} from "react";
import {searchService} from "../../../services/SearchService";
import {MagnifyingGlass} from "../../../assets/icons/Icons";
import {Button, Stack} from "react-bootstrap";

const HomeForm = () => {
    const [query, setQuery]: [query: string, setQuery: Dispatch<SetStateAction<string>>] = useState("");
    const [submit, setSubmit]: [submit: string, setSubmit: Dispatch<SetStateAction<string>>] = useState("abcde");
    const [errors, setErrors]: [errors: Array<string>, setErrors: Dispatch<SetStateAction<String[]>>] = useState([]);

    useEffect(() => {
        searchService(submit)
            .then(res => console.log(res))
    }, [submit])

    function handleChange({target}: { target: any }) {
        setQuery(target.value)
    }

    function handleSubmit() {
        if (query.length > 4) {
            setSubmit(query);
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <Stack gap={3}>
                <div className="search-bar">
                    <input type="text" onChange={handleChange} value={query}/>
                    <div className="submit">
                        <MagnifyingGlass size={28} color={"white"}/>
                    </div>
                </div>
                <Button className="submit-button">
                    <MagnifyingGlass size={28} color={"white"}/>
                </Button>
            </Stack>
        </form>
    )
}

export default HomeForm;