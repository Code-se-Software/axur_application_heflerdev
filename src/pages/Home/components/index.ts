export {Form} from "./Form"
export {Header} from "../../../components/Header"
export {Results} from "./Results"