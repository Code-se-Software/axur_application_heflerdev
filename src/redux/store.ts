import {configureStore} from '@reduxjs/toolkit'
import websiteReducer from "./slices/websiteSlice"

const store = configureStore({
    reducer: {
        website: websiteReducer,
    },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch