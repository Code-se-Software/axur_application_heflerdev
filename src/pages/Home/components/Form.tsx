import React, {Dispatch, SetStateAction, useEffect, useState} from "react";
import {searchService} from "../../../services";
import {MagnifyingGlass} from "../../../assets/icons/Icons";
import {Button, Stack} from "react-bootstrap";
import {useDispatch} from "react-redux";
import {AppDispatch} from "../../../redux/store";
import {insertData} from "../../../redux/slices/websiteSlice";

export const Form = () => {
    const [query, setQuery]: [query: string, setQuery: Dispatch<SetStateAction<string>>] = useState("");
    const [submit, setSubmit]: [submit: string, setSubmit: Dispatch<SetStateAction<string>>] = useState("abcde");
    const [errors, setErrors]: [errors: Array<string>, setErrors: Dispatch<SetStateAction<String[]>>] = useState([]);
    const dispatch: AppDispatch = useDispatch()

    useEffect(() => {
        searchService(submit)
            .then(res => dispatch(insertData(res.id)))
    }, [submit])

    function handleChange({target}: { target: any }) {
        setQuery(target.value)
    }

    function handleSubmit() {
        if (query.length > 4) {
            setSubmit(query);
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <Stack gap={3}>
                <div className="search-bar">
                    <input type="text" onChange={handleChange} value={query}/>
                    <div className="submit">
                        <MagnifyingGlass size={28} color={"white"}/>
                    </div>
                </div>
                <Button className="submit-button">
                    <MagnifyingGlass size={28} color={"white"}/>
                </Button>
            </Stack>
        </form>
    )
}