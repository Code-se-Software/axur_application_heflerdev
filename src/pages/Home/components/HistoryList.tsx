import React, {useEffect, useState} from "react";
import {Stack} from "react-bootstrap";
import {TrashIcon} from "src/assets/icons/Icons";
import ReactPaginate from "react-paginate";
import {useDispatch} from "react-redux";
import {AppDispatch} from "src/redux/store";
import {removeData, setData} from "src/redux/slices/websiteSlice";

interface Paginated {
    itemsPerPage: number,
    items: String[]
}

export const HistoryList = ({list}: { list: Array<{ id: string, keyword: string }> }) => {
    const [currentItems, setCurrentItems] = useState(null);
    const [pageCount, setPageCount] = useState(0);
    const [itemOffset, setItemOffset] = useState(0);
    const dispatch: AppDispatch = useDispatch()
    const itemsPerPage = 4;

    useEffect(() => {
        const endOffset = itemOffset + itemsPerPage;
        setCurrentItems(list.slice(itemOffset, endOffset));
        setPageCount(Math.ceil(list.length / itemsPerPage));
    }, [itemOffset, itemsPerPage, list]);

    const handlePageClick = (event: { selected: number; }) => {
        const newOffset = (event.selected * itemsPerPage) % list.length;
        setItemOffset(newOffset);
    };

    const cutKeyword = (keyword: string) => keyword.length > 8 ? keyword.slice(0, 8) + "..." : keyword;

    return (
        <Stack gap={1}>
            <div className="title-container">
                <div className="id">Id (Click to view)</div>
                <div>|</div>
                <div className="keyword">Search Term</div>
                <div>|</div>
                <div className="delete">Delete</div>
            </div>
            {currentItems &&
                currentItems.map(({id, keyword}: { id: string, keyword: string }, index: number) => (
                    <div key={id + index} className={"key-container"}>
                        <div className="id" onClick={() => dispatch((setData(id)))}>{id}</div>
                        <div className="keyword" title={keyword}>{cutKeyword(keyword)}</div>
                        <div className="icon" onClick={() => dispatch(removeData(id))}>
                            <TrashIcon/>
                        </div>
                    </div>
                ))
            }
            {list.length >= 5 &&
                <ReactPaginate pageCount={pageCount}
                               breakLabel="..."
                               nextLabel=">>"
                               onPageChange={handlePageClick}
                               pageRangeDisplayed={5}
                               previousLabel="<<"
                               renderOnZeroPageCount={null}
                               pageClassName="page-class-name"
                               pageLinkClassName="page-link"
                               previousClassName="previous-class-name"
                               previousLinkClassName="previous-link-class-name"
                               nextClassName="next-class-name"
                               nextLinkClassName="next-link-class-name"
                               breakClassName="break-class-name"
                               breakLinkClassName="break-link-class-name"
                               containerClassName="container-class-name"
                               activeClassName="active-class-name"
                />
            }
        </Stack>
    )
}