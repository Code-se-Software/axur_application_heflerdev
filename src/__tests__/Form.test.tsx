import React from "react";
import TestRenderer from "react-test-renderer";
import {Provider} from "react-redux";
import {store} from "src/redux/store";
import {Form} from "src/pages/Home/components";

describe("<Form>", () => {
    const renderer = TestRenderer.create(<Provider store={store}><Form/></Provider>)
    it("renders correctly", () => {
        expect(renderer.toJSON()).toMatchSnapshot()
    })
})