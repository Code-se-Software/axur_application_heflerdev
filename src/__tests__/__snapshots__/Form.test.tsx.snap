// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form> renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="vstack gap-3 minBreakpoint-xs"
  >
    <div
      className="search-bar"
    >
      <div
        className="error"
      >
        <span />
      </div>
      <input
        onChange={[Function]}
        placeholder="Need to have 4+ letters."
        type="text"
        value=""
      />
      <div
        className="submit"
        onClick={[Function]}
      >
        <span />
      </div>
    </div>
    <button
      className="submit-button btn btn-primary"
      disabled={false}
      type="submit"
    >
      <span />
    </button>
  </div>
</form>
`;
